### PR TITLE
NCSD-2860: Adds script to smoke test a given site

### PR DESCRIPTION
**Pull request by thelious1974**

### CODE REVIEW

File: PSCoreScripts/Invoke-SmokeTestOnWebApp.ps1 [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        [Parameter(Mandatory=$true)]
+        [string] $Url,
+        [Parameter(Mandatory=$true)]
+        [int] $TimeoutInSecs
+    )
+
+    try {
+        Write-Verbose "Performing Invoke-WebRequest on Uri '$Url'.."
+        $result = Invoke-WebRequest -Method Get -Uri $Url -MaximumRedirection 0 -TimeoutSec $TimeoutInSecs -UseBasicParsing
+        return $result.StatusCode -eq 200
```
Comment (NickGraham101) [inline PSCoreScripts/Invoke-SmokeTestOnWebApp.ps1]: Are we only considering 200s as a success or should other 2xx codes be included?

File: PSCoreScripts/Invoke-SmokeTestOnWebApp.ps1 [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        [Parameter(Mandatory=$false)]
+        [string] $Slot = "production",
+        [Parameter(Mandatory=$true)]
+        [string] $Path
+    )
+
+    $webApp = Get-AzWebAppSlot -ResourceGroupName $ResourceGroup -Name $AppName -Slot $Slot
+
+    if(-not $Path.StartsWith("/")) {
+        $Path = "/$($Path)"
```
Comment (rivoric) [inline PSCoreScripts/Invoke-SmokeTestOnWebApp.ps1]: The string seems overly complex, why not $Path = "/$Path"

File: PSCoreScripts/Invoke-SmokeTestOnWebApp.ps1 (hunk 1 above)
Comment (NickGraham101) [reply PSCoreScripts/Invoke-SmokeTestOnWebApp.ps1]: Might be worth being explicit about what is considered a successful smoke test in the description

File: Tests/powershellcore/U033.PSCoreScripts.Invoke-SmokeTestOnWebApp.Tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,185 @@
+﻿Push-Location -Path $PSScriptRoot\..\..\PSCoreScripts\
+
+Describe "Invoke-SmokeTests unit tests" -Tag "Unit" {
+
+    function Invoke-WebRequest { param($Method, $Uri, $MaximumRedirection, $TimeoutSec, [switch] $UseBasicParsing) }
```
Comment (rivoric) [inline Tests/powershellcore/U033.PSCoreScripts.Invoke-SmokeTestOnWebApp.Tests.ps1]: Do we need this. I thought Invoke-WebRequest (like Start-Sleep) was part of the core not loaded from a module

File: PSCoreScripts/Invoke-SmokeTestOnWebApp.ps1 (hunk 1 above)
Comment (thelious1974) [reply PSCoreScripts/Invoke-SmokeTestOnWebApp.ps1]: I've expanded this to be any 2xx status code, and expanded the documentation to explicitly say this, and to say that it does not follow any redirects.

File: Tests/powershellcore/U033.PSCoreScripts.Invoke-SmokeTestOnWebApp.Tests.ps1 (hunk 3 above)
Comment (thelious1974) [reply Tests/powershellcore/U033.PSCoreScripts.Invoke-SmokeTestOnWebApp.Tests.ps1]: We didn't - and the Az module allows us to properly mock Get-AzWebAppSlot, so I've also removed the function definition for that too.

**Pull request merged**